CRM/Contribute/BAO/Contribution.php

```diff
@@ -955,8 +955,9 @@ public static function recordPaymentActivity($contributionId, $participantId, $t
    * @return int
    */
   public static function getToFinancialAccount($contribution, $params) {
-    if (!empty($params['payment_processor'])) {
-      return CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($contribution['payment_processor'], NULL, 'civicrm_payment_processor');
+    $paymentProcessorID = $params['payment_processor_id'] ?? CRM_Utils_Array::value('payment_processor', $params);
+    if (!empty($paymentProcessorID)) {
+      return CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($paymentProcessorID, NULL, 'civicrm_payment_processor');
```

Comment:
@eileenmcnaughton There is a change here.  Previously this line was getting the payment processor ID from the `$contribution` array and now it's getting it from the `$params` array.

Reply:
Yeah - I think it's more correct but I'm going to write a test & do a bit more work on this.  I'm targetting the 5.19 rc but am not going to try to do a 5.18 drop on this - I think that's the right level of 'rush'